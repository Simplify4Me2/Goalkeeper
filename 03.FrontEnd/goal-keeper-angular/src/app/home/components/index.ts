export * from './match-day/match-day.component';
export * from './news-item/news-item.component';
export * from './ranking/ranking.component';