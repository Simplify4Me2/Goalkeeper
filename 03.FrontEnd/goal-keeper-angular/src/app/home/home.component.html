<p>home works!</p>

<h1>Fixture</h1>
<div class="flex border" *ngFor="let fixture of fixtures" >
    <span>{{ fixture.homeTeam }}</span>
    <div>{{ fixture.homeScore }} - {{ fixture.awayScore }}</div>
    <span>{{ fixture.awayTeam }}</span>
</div>
