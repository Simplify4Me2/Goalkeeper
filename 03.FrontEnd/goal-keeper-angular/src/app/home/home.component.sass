.flex
    display: flex

span 
    padding: 0px 5px