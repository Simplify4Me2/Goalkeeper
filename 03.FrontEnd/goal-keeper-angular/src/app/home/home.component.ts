import { Component, OnInit } from '@angular/core';
import { Fixture } from './fixture.model';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.sass'],
})
export class HomeComponent implements OnInit {
  fixtures: Fixture[] = [
    { homeTeam: 'OHL', homeScore: 0, awayTeam: 'Anderlecht', awayScore: 1 },
    { homeTeam: 'Club Brugge', homeScore: 4, awayTeam: 'Zulte Waregem', awayScore: 1 },
  ];

  constructor() {}

  ngOnInit(): void {}
}
