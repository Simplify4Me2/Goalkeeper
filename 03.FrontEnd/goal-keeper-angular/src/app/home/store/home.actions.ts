import { createAction, props } from '@ngrx/store';
import { Fixture } from '../models/fixture.model';
import { Ranking } from '../models/ranking.model';

export const getFixtures = createAction(
    '[Home/Container] Get Fixtures'
);

export const getFixturesSuccess = createAction(
    '[Home/API] Get Fixtures Success',
    props<{ fixtures: Fixture[] }>()
);

export const getFixturesFail = createAction(
    '[Home/API] Get Fixtures Success',
);

export const getRanking = createAction(
    '[Home/Container] Get Ranking'
);

export const getRankingSuccess = createAction(
    '[Home/API] Get Ranking Success',
    props<{ ranking: Ranking }>()
);

export const getRankingFail = createAction(
    '[Home/API] Get Ranking Success',
);