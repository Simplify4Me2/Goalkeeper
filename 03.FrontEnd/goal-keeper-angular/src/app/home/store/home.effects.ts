import { Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { of } from 'rxjs';
import { catchError, map, switchMap } from 'rxjs/operators';

import { Fixture } from '../models/fixture.model';
import { Ranking } from '../models/ranking.model';
import * as fromActions from './home.actions';
import { FixturesService } from './services/fixtures.service';
import { RankingService } from './services/ranking.service';

@Injectable()
export class HomeEffects {

    constructor(private actions: Actions, private fixturesService: FixturesService, private rankingService: RankingService) { }

    getFixtures = createEffect(() =>
        this.actions.pipe(
            ofType(fromActions.getFixtures),
            switchMap(() =>
                this.fixturesService.get().pipe(
                    map((fixtures: Fixture[]) => fromActions.getFixturesSuccess({ fixtures })),
                    catchError(() => of(fromActions.getFixturesFail()))
                )
            )
        )
    );

    getRanking = createEffect(() =>
        this.actions.pipe(
            ofType(fromActions.getRanking),
            switchMap(() =>
                this.rankingService.deprecatedGet().pipe(
                    map((ranking: Ranking) => fromActions.getRankingSuccess({ ranking })),
                    catchError(() => of(fromActions.getRankingFail()))
                )
            )
        )
    );
}