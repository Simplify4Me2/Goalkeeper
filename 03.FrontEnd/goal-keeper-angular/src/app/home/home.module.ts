import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { HomeRoutingModule } from './home-routing.module';
import { FixtureComponent } from './components/fixture/fixture.component';
import { HomeComponent } from './containers/home.container';
import * as fromHome from './store';
import { HomeEffects } from './store/home.effects';
import { RankingComponent } from './components/ranking/ranking.component';
import { NewsItemComponent } from './components/news-item/news-item.component';

@NgModule({
    declarations: [
        HomeComponent, FixtureComponent, RankingComponent, NewsItemComponent
    ],
    imports: [
        CommonModule,
        HomeRoutingModule,
        StoreModule.forFeature(fromHome.homeFeatureKey, fromHome.reducer),
        EffectsModule.forFeature([HomeEffects]),
    ],
})
export class HomeModule {}