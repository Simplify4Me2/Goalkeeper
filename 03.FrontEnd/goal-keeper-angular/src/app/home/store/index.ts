import {
  Action,
  createFeatureSelector,
  createReducer,
  createSelector,
  on,
} from '@ngrx/store';

import * as fromRoot from '../../store';
import { Ranking } from '../models/ranking.model';
import * as fromActions from './home.actions';
import { Matchday } from '../models/matchday.model';

import moment from 'moment';

export const homeFeatureKey = 'home';

export interface HomeState {
  ranking: Ranking;
  matchday: Matchday;
}

export const initialState: HomeState = {
  ranking: null,
  matchday: null,
};

export interface State extends fromRoot.State {
  [homeFeatureKey]: HomeState;
}

const testReducer = createReducer(
  initialState,
  on(fromActions.getMatchdaySuccess, (state, { matchday }) => ({
    ...state,
    matchday: matchday,
  })),
  on(fromActions.getRankingSuccess, (state, { ranking }) => ({
    ...state,
    ranking: ranking,
  }))
);

export function reducer(state: HomeState | undefined, action: Action) {
  return testReducer(state, action);
}

export const selectHomeState =
  createFeatureSelector<State, HomeState>(homeFeatureKey);

// export const selectFixtures = createSelector(
//   selectHomeState,
//   (state) => state.fixtures
// );
export const selectRanking = createSelector(
  selectHomeState,
  (state) => state.ranking
);

export const selectMatchday = createSelector(selectHomeState, (state) => state.matchday);

// export const selectMatchday = createSelector(selectHomeState, (state) => ({
//   ...state.matchday,
//   matches: state.matchday?.matches.map((match) => ({
//     ...match,
//     date: moment(match.date).format('DD/MM'),
//   })),
// }));
