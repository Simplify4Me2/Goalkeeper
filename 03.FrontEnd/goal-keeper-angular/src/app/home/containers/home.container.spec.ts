import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { MockStore, provideMockStore } from '@ngrx/store/testing';

import * as fromStore from '../store';
import * as fromActions from '../store/home.actions';
import { HomeComponent } from '../containers/home.container';
import { MatchDayComponent, RankingComponent } from '../components';
import { SharedModule } from '../../shared/shared.module';

describe('Home Container', () => {
    let fixture: ComponentFixture<HomeComponent>;
    let instance: HomeComponent;
    let store: MockStore;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                SharedModule,
                RouterTestingModule
            ],
            declarations: [
                HomeComponent,
                RankingComponent,
                MatchDayComponent
            ],
            providers: [
                provideMockStore({
                    selectors: [{ selector: fromStore.selectLastMatchday, value: [] }, { selector: fromStore.selectRanking, value: [] },],
                }),
            ],
        });

        fixture = TestBed.createComponent(HomeComponent);
        instance = fixture.componentInstance;
        store = TestBed.inject(MockStore);

        spyOn(store, 'dispatch');
    });

    it('should compile', () => {
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should dispatch a getRankings on init', () => {
        const action = fromActions.getRanking();

        fixture.detectChanges();

        expect(store.dispatch).toHaveBeenCalledWith(action);
    });
});