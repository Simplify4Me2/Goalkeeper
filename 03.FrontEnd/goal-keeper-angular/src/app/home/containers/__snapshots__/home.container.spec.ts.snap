// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container should compile 1`] = `
<app-home
  fixtures={[Function Store]}
  ranking={[Function Store]}
  store={[Function MockStore]}
>
  <h2
    class="dashboard-title"
  >
    Soccer stats insights for your favourite team
  </h2><div
    class="dashboard-overview"
  >
    <section
      class="main-content"
    >
      <article>
        <h1>
          Fixtures
        </h1>
        
      </article>
      <article>
        <h1>
          News
        </h1>
        <app-news-item>
          <p>
            news-item works!
          </p>
          <picture>
            <img
              src="/assets/news-item.jpg"
            />
          </picture>
        </app-news-item>
      </article>
    </section>
    <aside
      class="side-content"
    >
      <h1>
        Ranking
      </h1>
      <app-ranking>
        <table
          class="ranking-table"
        >
          <thead>
            <tr>
              <th>
                Position
              </th>
              <th>
                Team
              </th>
              <th>
                Points
              </th>
            </tr>
          </thead>
          
          <tbody>
            
          </tbody>
        </table>
      </app-ranking>
    </aside>
  </div>
</app-home>
`;
